netket/nn/symmetric_linear.py

```diff
@@ -242,41 +227,48 @@ def __call__(self, x: Array) -> Array:
 
         dtype = jnp.promote_types(x.dtype, self.dtype)
         x = jnp.asarray(x, dtype)
+        # Infer in_features and add batch and in_feature dimensions as necessary
+        if x.ndim == 1:
+            in_features = 1
+            x = jnp.expand_dims(x, (0, 1))
+        elif x.ndim == 2:
+            in_features = 1
+            x = jnp.expand_dims(x, 1)
```

Comment:
I think we should throw a warning to the user saying that we are upscaling the input
and remove this in a very far future